pkg/imgutil/dockerconfigresolver/dockerconfigresolver.go

```diff
@@ -0,0 +1,122 @@
+/*
+   Copyright (C) nerdctl authors.
+   Copyright (C) containerd authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package dockerconfigresolver
+
+import (
+	"net/url"
+
+	"github.com/containerd/containerd/remotes"
+	"github.com/containerd/containerd/remotes/docker"
+	dockercliconfig "github.com/docker/cli/cli/config"
+	dockercliconfigtypes "github.com/docker/cli/cli/config/types"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// New instantiates a resolver using $DOCKER_CONFIG/config.json .
```

Comment:
@fuweid @ktock
Do you know whether there is an existing package for this?

Reply:
Unfortunately, don't see the package 😂

Reply:
Me too.